### PR TITLE
Ensure outdated templates are removed

### PR DESCRIPTION
**Pull request by deeplow**
## Status

Ready for review

## Description of Changes

Inverts the logic of test_expected from checking that the VMs we wanted do exist...
https://github.com/freedomofpress/securedrop-workstation/blob/23529537b484d801ad921e08cc43ee867fce5549/tests/test_vms_exist.py#L22-L25

...to one where we account for all the VMs tagged with `sd-workstation` or any other VM that we may have missed.

## Testing

This is a test already.

## Deployment

No deployment considerations.

### If you have made changes to the provisioning logic

- [x] All tests (`make test`) pass in `dom0`

### If documentation is required
Docs not required.

### CODE REVIEW

**Comment by cfm** [comment]
(Assigning per Slack discussion just for visibility; feel free to reassign as needed.)

File: tests/test_vms_exist.py [hunk 1]
```diff
@@ -20,17 +25,17 @@ def tearDown(self):
         pass
 
     def test_expected(self):
-        vm_set = set(self.app.domains)
-        for test_vm in WANTED_VMS:
-            self.assertIn(test_vm, vm_set)
+        sdw_tagged_vm_names = [vm.name for vm in self.sdw_tagged_vms]
+        expected_vms = set(SD_VMS + SD_DVM_TEMPLATES + SD_TEMPLATES)
+        self.assertEqual(set(sdw_tagged_vm_names), set(expected_vms))
```
Comment (rocodes) [inline tests/test_vms_exist.py:30]: Just noting for our future selves: Sometimes the set of sd-tagged vms isn't exactly equal to the set of VMs expressed here, but they're all edge cases and shouldn't interfere:
- devs may tag their sd-dev vm @tag:sd-workstation if they follow our (old?) developer guidelines. This shouldn't be a problem though because those environments aren't where we should be running `make test` since they're not indicative of real conditions
- prod instances may(?) configure additonal VMs and tag them with @sd-workstation to take advantage of our strict rpc policies. I don't think this is happening, but in any case, prod setups shouldn't be running dom0 tests.

**Review by rocodes — APPROVED** [review]
Visual review LGTM, CI LGTM, laughed at 

> ## Testing
> This is a test already.

:) 

Thanks deeplow. I think this fixes #1091 too.
